### ISSUE TITLE
SC2154: Constant within ((expression)) is flagged as referenced by not assigned

### ISSUE THREAD

**Issue by Chubler-XL** [koalaman/shellcheck#928]
#### For bugs
- Rule Id: SC2154
- My shellcheck version 0.3.7
- I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

#!/bin/bash
read -sr -n1 action
((action == "q")) && exit 0

#### Here's what shellcheck currently says:

Line 3:
((action == "q")) && exit 0
            ^-- SC2154: q is referenced but not assigned.

#### Here's what I wanted or expected to see:
"q" is a constant, not a variable.  It does not need to be assigned.

**Comment by Nightfirecat** [comment]
I can repro on v0.4.6 and on shellcheck.net.

**Comment by koalaman** [comment]
ShellCheck is right. `"q"` refers to the variable named `q` in arithmetic expressions:

```
$ action=1 q=1
$ (( action == "q" )) && echo "same" || echo "different"
same
$ action=1 q=2
$ (( action == "q" )) && echo "same" || echo "different"
different
```

There should probably be a warning saying that quoted strings are variable names though. Strings should be compared with `[` or `[[`
